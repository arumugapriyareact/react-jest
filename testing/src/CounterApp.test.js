import React from "react";
import { shallow } from "enzyme";
import CounterApp from "./CounterApp";
import renderer from "react-test-renderer";

describe("Counter component", () => {
  it("starts with a count of 0", () => {
    const wrapper = shallow(<CounterApp />);
    const text = wrapper.find("p").text();
    expect(text).toEqual("Count: 0");
  });
});

describe("test button click", () => {
  it("on click button", () => {
    const mockCallBack = jest.fn();
    // let count = 0;
    // const inc = () => {
    //   count++;
    // };
    const button = shallow(
      <button className="Increment" onClick={mockCallBack}>
        Increment count
      </button>
    );
    button.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
    // expect(count).toEqual(0);
  });
});

describe("decrement the count", () => {
  it("decrement value", () => {
    const wrapper = shallow(<CounterApp />);
    const decbtn = wrapper.find("button.Decrement");
    decbtn.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual("Count: -1");

    const mockCallBack = jest.fn();
    const decBtnClick = shallow(
      <button className="Decrement" onClick={mockCallBack}>
        Decrement count
      </button>
    );
    decBtnClick.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});

describe("test snap", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<CounterApp />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
