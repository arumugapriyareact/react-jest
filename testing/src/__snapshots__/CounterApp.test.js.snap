// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snap matches the snapshot 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    className="Increment"
    onClick={[Function]}
  >
    Increment count
  </button>
  <button
    className="Decrement"
    onClick={[Function]}
  >
    Decrement count
  </button>
</div>
`;
